src/esssans/loki/masking.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+"""
+Masking functions for the loki workflow.
+"""
+from typing import NewType, Optional
+
+import numpy as np
+import scipp as sc
+
+from ..types import (
+    BeamStopPosition,
+    BeamStopRadius,
+    MaskedData,
+    RawData,
+    RunType,
+    SampleRun,
+)
+
+DetectorLowCountsStrawMask = NewType('DetectorLowCountsStrawMask', sc.Variable)
+"""Detector low-counts straw mask"""
+DetectorBeamStopMask = NewType('DetectorBeamStopMask', sc.Variable)
+"""Detector beam stop mask"""
+DetectorTubeEdgeMask = NewType('DetectorTubeEdgeMask', sc.Variable)
+"""Detector tube edge mask"""
+
+
+def detector_straw_mask(
+    sample_straws: RawData[SampleRun],
+) -> DetectorLowCountsStrawMask:
+    return DetectorLowCountsStrawMask(
+        sample_straws.sum('pixel').data
+        < sc.scalar(sample_straws.sizes['pixel'] * 0.5, unit='counts')
+    )
```

Comment:
Still worried about such masks, we should never simply mask zero or low count pixels!

Reply:
We have such masks in [SANS2D](https://github.com/scipp/esssans/blob/main/src/esssans/sans2d.py#L93), so you would also want to change it there?

I agree that masking individual pixels based on counts is not good, for the reasons you stated.
However, the mask here is masking straws that have very few counts when summed over the pixel dimension, which seems to work well for finding dead straws. I have checked the results using different visualizations.

I am open to suggestions as to how we could achieve this otherwise, but right now the alternative I see is to have hard-coded numbers for the y positions of the pixels (e.g. `if y > 0.55 * meter or y < 0.45 * meter`), which would have to change for each run if the beam position has changed?

Reply:
> We have such masks in [SANS2D](https://github.com/scipp/esssans/blob/main/src/esssans/sans2d.py#L93), so you would also want to change it there?

That is slightly different because it combines a spatial condition with a counts condition. It may still be questionable though.


> I agree that masking individual pixels based on counts is not good, for the reasons you stated. However, the mask here is masking straws that have very few counts when summed over the pixel dimension, which seems to work well for finding dead straws. I have checked the results using different visualizations.

Wouldn't that also risk masking straws that are far from the beam center? How did you determine the threshold?

> I am open to suggestions as to how we could achieve this otherwise, but right now the alternative I see is to have hard-coded numbers for the y positions of the pixels (e.g. `if y > 0.55 * meter or y < 0.45 * meter`), which would have to change for each run if the beam position has changed?

Supposedly dead components will be masked in the NeXus file.